core/vm/evm.go

```diff
@@ -266,6 +356,65 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 			contract.UseGas(contract.Gas)
 		}
 	}
+
+	if UsingOVM {
```

Comment:
What is the rationale behind the whole isTarget/OriginalTarget logic? What can go wrong here?